include/circt/Dialect/Sim/SimOps.td

```diff
@@ -52,6 +52,29 @@ def FatalOp : SimOp<"fatal"> {
   let results = (outs);
 
   let assemblyFormat = "$clk `,` $cond attr-dict";
+
+}
+
+def PrintOp : SimOp<"print"> {
+  let summary = "Prints a formatted string to `stderr`";
+
+  let description = [{
+    Prints a formatted string, using standard SystemVerilog placeholders,
+    to `stderr` on the rising edge of `clk` when `cond` is high.
+  }];
+
+  let arguments = (ins
+    ClockType:$clk,
+    I1:$cond,
+    StrAttr:$format_string,
+    Variadic<AnyType>:$substitutions
+  );
+  let results = (outs);
+
+  let assemblyFormat = [{
+    $clk `,` $cond `,` $format_string attr-dict (`(` $substitutions^ `)`
+      `:` qualified(type($substitutions)))?
+  }];
```

Comment:
Could this use the Verif dialect's [FormatVerilogStringOp](https://github.com/llvm/circt/blob/main/include/circt/Dialect/Verif/VerifOps.td#L46-L78) instead of doing formatting? I think the consensus there was to try factor formatting out into one op, instead of having many ops deal with the weird SV formatting.

I guess using Verif's `PrintOp` is not really posisble here, since you want an op with a clock that you can put in a graph region, right?